MCP-7609 Updating GEMs script

LauraJohnson-1 opened the pull request:
# Feature Name
GEMs script updates

## Description
<!-- you don't need to write anything here -->

### What was the problem?
Previous GEMs script required an input file of only ICD-10 codes and ICD-10 descriptions. The previous output included duplicate ICD-9 codes and multiple columns of flags. VASRD codes and CFR criteria would need to be added to all ICD-9 codes by the user before adding to the code set.

### How does this fix it?
These updates allow the code set file with the "Code Set Details" sheet to be used as the input. VASRD codes and CFR Criteria, mapped from the ICD-10 codes, are included in the output. The output also simplifies review by removing duplicate ICD-9 codes, concatenating ICD-10 codes and descriptions for the codes that need further review and formats the output allowing it to be copy and pasted into the code set once codes are reviewed.

### Jira Tickets

- [MCP-7609](https://amida.atlassian.net/browse/MCP-7609)

## How to test this PR

1. Run script using any condition's code set file
2. Review output, specifically rows that have the "Approximate" or "Requires combination" flag
3. Review README file

## Definition of Done Checklist

- [x] Code follows the project's coding style guidelines
- [x] Linting complete (Pylint score: 10/10)
- [x] Any disabled linting rules are explained in the description
- [x] Relevant documentation has been updated
- [x] Tests have been written and passed
- [x] No console errors or warnings

[MCP-7609]: https://amida.atlassian.net/browse/MCP-7609?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

jlpeterson2 commented on GEMs/icd10_to_icd9_backward_mapping.py:
```diff
@@ -0,0 +1,109 @@
+"""
+This script will reverse map ICD-9 codes from the GEMs files to ICD-10 codes in the input file
```
Should read something like this (switch ICD-9 & ICD-10): `This script will reverse map ICD-10 codes to ICD-9 codes using the GEMs files`

LauraJohnson-1 replied:
Updated!

jlpeterson2 replied:
Thanks! Can you double check you committed this update? I'm not seeing it yet.

jlpeterson2 commented on GEMs/icd10_to_icd9_backward_mapping.py:
```diff
@@ -0,0 +1,109 @@
+ICD9codes = ICD9codes[ICD9codes['Flag'].notna()]
+
+
+def flag_review(i):
```
Wonderful job at combining these! Love it!

jlpeterson2 commented on GEMs/icd10_to_icd9_backward_mapping.py:
```diff
@@ -0,0 +1,109 @@
+# an "Approximate" or "Requires Combination" Flag is necessary.
+
+OUTPUT_FILE_PATH = f"output/{CODE_SET_NAME}_confirmed.xlsx"
+CombinedICD9.to_excel(OUTPUT_FILE_PATH, index=False)
```
Can we add an indicator that the file was saved. Similar to the API scripts: 
`# Print a message indicating where the file is saved
print(f"Excel file '{CODE_SET_NAME{' saved in the output folder.")`

jlpeterson2 commented:
@LauraJohnson-1 This works wonderful, thank you! I left a few comments. In addition: 
- Can we rename the python file to have a verb in front of the name. For example: `run_icd10_to_icd9_backward_mapping` or `map_icd10_to_icd9_codes` 
- Is there a reason the accurate matches do not include the corresponding ICD-10 codes? I know a user won't need them to double check, but curious if it would add relevant context just in case. Thoughts? 

Thanks!

ZubairL commented on GEMs/icd10_to_icd9_backward_mapping.py:
```diff
@@ -0,0 +1,109 @@
+ICD9codes = ICD9codes.iloc[:, [0, 1, 2, 11, 13, 12, 3, 4]]
+
+# Remove rows where ICD9 or Flag columns are blank
+ICD9codes = ICD9codes[ICD9codes['ICD9'].notna()]
```
Getting a keyerror here:

Traceback (most recent call last):
  File "c:\Users\standard\Desktop\MCP_ADS\ADS\GEMs\icd10_to_icd9_backward_mapping.py", line 54, in <module>
    ICD9codes = ICD9codes[ICD9codes['ICD9'].notna()]
                          ~~~~~~~~~^^^^^^^^
  File "C:\Users\standard\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\pandas\core\frame.py", line 4102, in __getitem__
    indexer = self.columns.get_loc(key)
              ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\standard\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\pandas\core\indexes\base.py", line 3812, in get_loc
    raise KeyError(key) from err
KeyError: 'ICD9'



I only seem to get this with a few files( eye and mental health worked great) and per our message in slack it's due to calling out index position of columns and not column names. Leaving here just for documentation and in case anyone else gets this error.

ZubairL reviewed (changes requested):
Could you also put an example Code Set file and remove the old Test_gi-icd10.xlsx file?

LauraJohnson-1 commented:
Thank you @ZubairL and @jlpeterson2 for your reviews! 
I have made all suggested changes. I changed around the column filtering (was by index previously, not it is by column name), and removed the `pd.concat` due to including the ICD-10 codes for the accurate codes)
Let me know if you run into anything else!

ZubairL reviewed (approved):
Nice, looks good!

jlpeterson2 commented on GEMs/map_icd10_to_icd9_codes.py:
```diff
@@ -0,0 +1,101 @@
+AddICD9 = pd.merge(AddICD9, ICD9desc, on=['ICD9'], how='left')
+
+# Reformat ICD9 and ICD10 codes, names
+AddICD9['ICD9'] = (AddICD9['ICD9'].str[:3] + '.' + AddICD9['ICD9'].str[3:])
```
Can you remove the parenthesis for these two lines? It is pulling up in my pylint check. 

```
AddICD9['ICD9'] = AddICD9['ICD9'].str[:3] + '.' + AddICD9['ICD9'].str[3:]
AddICD9['ICD10'] = AddICD9['ICD10'].str[:3] + '.' + AddICD9['ICD10'].str[3:]
```

jlpeterson2 reviewed (approved):
Thanks for the quick updates! Looks good!
